libqtile/state.py

```diff
@@ -45,6 +53,39 @@ def apply(self, qtile):
             Rearrange the windows in the specified Qtile object according to
             this QtileState.
         """
+        try:
+            for group in qtile.groups:
+                x = []
+                for i in self.focusHistory[group.name]:
+                    x.append(qtile.windowMap[i.Wid])
+                group.focusHistory = x
+                for layout in group.layouts:
```

Comment:
If I use pickling only, some of the attributes always get lost. For example in `window` class, it has an attribute group which has a link back to the window. This is a recursive chain , so  during `__getstate__` I pickled every attribute of the window class but not it's group.However , after unpickling many of the attributes of the window class became *unset*. Removing the `group` attribute causes a lot of changes to the `window` state and so as it wasn't possible for me to restore the whole state back to original by just using picklng alone. I therefore only stored the `wid` during pickling and used it to get the window information back later.